docs(README): Update README structure (#4675)
liferay/clay: README.md

<h1 align="center">
    <div>
        <img src="https://github.com/liferay/clay/blob/6ae276a67e7e2c9954d72dcae15266b2458f58f9/clayui.com/static/images/clay_logo_w.png?raw=true"> 
    </div>
    <a href="https://clayui.com/">
        Clay
    </a>

[![Build Status](https://travis-ci.org/liferay/clay.svg?branch=master)](https://travis-ci.org/liferay/clay) [![Coverage Status](https://coveralls.io/repos/github/liferay/clay/badge.svg)](https://coveralls.io/github/liferay/clay)

> </h1>

Author: Yeah I've understood it. I agree with you, it's better to determine first a ´<div>´ instead of a child element 👍

> </h1>

Reviewer: Great

> Clay is Liferay's web implementation of [Lexicon Experience Language](https://liferay.design/lexicon). Built with Bootstrap as a foundation, it's comprised of HTML, CSS, and JS.

Reviewer: ```suggestion
Clay is Liferay's web implementation of the [Lexicon Experience Language](https://liferay.design/lexicon). Built with Bootstrap as a foundation, it's comprised of HTML, CSS, and JS.
```

![Chrome Android](https://img.shields.io/badge/Chrome_Android-Latest-green.svg?style=flat)
![Safari iOS Latest](https://img.shields.io/badge/Safari_iOS-Latest-green.svg?style=flat)

## Packages

Clay repository follows the monorepo approach, all the major components that are the pillars of Clay are here in this repository.

> 1. ### [Clay Css](./packages/clay-css): The pillar to give colors and structure to Clay, where is the css and examples of HTML markups of the components.

Reviewer: ```suggestion
1. **[Clay Css]**(./packages/clay-css): The pillar to give colors and structure to Clay, where is the css and examples of HTML markups of the components.
```
Reason: I don't think we need to use an `h2` here because it looks "big", bold should be enough

2. ### [Clay React Components](./packages): All components that follow clay-css marking and Lexicon use cases, developed on [React.js](http://reactjs.org)

> 3. ### [Clayui.com](./clayui.com): Contains the source code and documentation powering [clayui.com](https://clayui.com/).

Reviewer: Same for this one

You can improve by sending pull requests to this repository.

## Migration Guides

Do you already have the Clay v2 components implemented in your project? These handy guides will help you perform the migration from [Clay v2](https://v2.clayui.com) to [v3](https://clayui.com).

-   [Migrate the Clay components from v2 to v3](https://clayui.com/docs/get-started/migrate-the-clay-components-from-v2-to-v3.html)

## Installation

> Clay is available as the `clay` package in npm for components and for css like `clay-css`. It is also available on a [CDN](https://cdn.jsdelivr.net/npm/clay/lib/js/clay.js) for components and for css, [check](https://cdn.jsdelivr.net/npm/clay/lib/css/atlas.css). But if you just want the separate components, it's also available in the npm as [clay-](https://www.npmjs.com/search?q=clay-).

Reviewer: The `clay` package in npm, is an older version of Clay as far as I can remember. 
Maybe we don't need to mention this (I wouldn't). Also, and just so you know and since this isn't easy to understand as someone "new" on the team: 

- In clay `2.x`, we used Metal.js and Soy and published the components with the following naming convention: `clay-COMPONENT_NAME`, for example: `clay-button`
- In clay `3.x`, we now use React (and TypeScript) and packages are published with the following naming convention: `@clayui/COMPONENT_NAME`, for example: `@clayui/button`

Author: If this point (Installation) explains `clay` npm package and it is an old version that someone new doesn't understand, I have two choices:

- Delete **Installation** point because it explains several concepts that are not easy to understand.
- In **Migration Guides** point I can add a small section with some information about clay `2.x` and clay `3.x`.

What do you think @julien?

Reviewer: I would go for 

> Delete Installation point because it explains several concepts that are not easy to understand.

Simply because no one is going to understand :rofl: 

Clay is available as "various" packages, there's actually one package per component, although that might change in the near future (see #4191 when you have time) 

Since this is the 3.x branch, I'd avoid mentioning anything about 2.x anyway, since it's just there for maintenance purposes or security fixes.

On top of that, anyone familiar with npm/React should be able to see very easily what the name (in the npm registry) of a component is by looking at the corresponding package.json file in the component's directory.

See the [Getting Started](https://clayui.com/docs/get-started/index.html) guide for more information.

## Setup

-   #### **Packages**

> To contribute to Clay React Components, you can try to run locally to test your changes:

Reviewer: I'm not sure you need the "bold" there

> 1. Install NodeJS >= [v6.11.0](http://nodejs.org/dist/v6.11.0/), if you don't have it yet.

Reviewer: I would suggest, installing the latest "LTS" version, that way we don't need to update this every 6 weeks

Author: I'll change the Node version to 16.x, it seems that we are updated 😄

Reviewer: @sergiojimcos, I was actually saying that you shouldn't mention any specific version: clay should work with whatever LTS version is available. Don't worry I've fetched the branch and will push more stuff on top of your changes.

> yarn run jest --u

Reviewer: `yarn run jest -u` is only needed to run the tests and update the snapshots.

```

> -   #### **Clayui.com**

Reviewer: I'm not sure you need this to be capitalized since it's an actual URL, and I don't think it needs to be in bold, since you're already using `####`.

To contribute to the documentation and the site in general, you can try to run locally to test your changes:

> 1. Move to the site folder:

Reviewer: ```suggestion
Navigate to the site directory and install the dependencies
```

yarn
```

3. Run in development environment:

```
yarn develop
```

> You can run clayui.com on Local LAN:

```
yarn develop -H 0.0.0.0
```

> > If you want to test in a production environment so that you do not take risks of inconsistencies, in the root folder:

Reviewer: > If you want to test in a production environment so that you do not take risks of inconsistencies, issue the following command in the root directory:

I prefer using "directory" over "folder", because that's usually how they are defined in operating systems and documentation

```
yarn site
```

## Contributing

> Feel free to open up problems or send pull requests. We will always be looking at these problems and we will be responding whenever possible.

Reviewer: Great addition, I'd just change it to 

```suggestion
Feel free to create issues or submit pull requests. Clay is actively maintained and your contributions are always welcome.
```
